v2/internal/reconcilers/annotations.go

```diff
@@ -6,13 +6,17 @@ Licensed under the MIT license.
 package reconcilers
 
 import (
+	"reflect"
+
 	"github.com/go-logr/logr"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 
```

Comment:
Note to self... we might be able to write a test that forces this to happen... maybe should do that in this PR.

Reply:
I thought about this some more and actually, rather than trying to force this in and end-to-end test, which is fragile and sensitive to timing, really what we need is just another `interval_calculator` unit test which ensures that we have the right backoff when we hit conflict.

I've added that unit test.